File: app/src/main/java/com/github/fo2rist/mclaren/dagger/McLarenFeedFragmentModule.java
package com.github.fo2rist.mclaren.dagger;

import com.github.fo2rist.mclaren.analytics.EventsLoggerImpl;
import com.github.fo2rist.mclaren.mvp.FeedContract;
import com.github.fo2rist.mclaren.repository.RepositoryEventBusImpl;
import com.github.fo2rist.mclaren.repository.McLarenFeedRepositoryImpl;
import com.github.fo2rist.mclaren.ui.feedscreen.FeedPresenter;
import dagger.Module;
import dagger.Provides;

@Module
class McLarenFeedFragmentModule {
    @Provides
    @Scopes.PerFragment
    FeedContract.Presenter providePresenter(
            McLarenFeedRepositoryImpl repository,
            RepositoryEventBusImpl eventBus,
            EventsLoggerImpl logger
    ) {
        return new FeedPresenter(repository, eventBus, logger);
    }
}

File: app/src/main/java/com/github/fo2rist/mclaren/dagger/McLarenFeedFragmentModule.kt
package com.github.fo2rist.mclaren.dagger

import com.github.fo2rist.mclaren.mvp.FeedContract
import com.github.fo2rist.mclaren.repository.feed.FeedRepository
import com.github.fo2rist.mclaren.ui.feedscreen.FeedPresenter
import com.github.fo2rist.mclaren.ui.feedscreen.McLarenFeedFragment
import dagger.Binds
import dagger.Module
import javax.inject.Named

@Module
internal interface McLarenFeedFragmentModule {

    @Binds
    @Scopes.PerFragment
    fun provideMcLarenFeedView(view: McLarenFeedFragment): FeedContract.View

    @Binds
    @Scopes.PerFragment
    fun provideMcLarenFeedRepository(@Named("mclaren-feed") repository: FeedRepository): FeedRepository

    @Binds
    @Scopes.PerFragment
    fun providePresenter(presenter: FeedPresenter): FeedContract.Presenter
}
